[WIP] Export cap

noahmanger opened the pull request:
Adds logic to disable the export button if the table has over 100,000 results. Hovering over or focusing on the button shows a tooltip.

If the table has under 100,000 results, it enables the button and the tooltip never shows.

![demo](http://g.recordit.co/XKlH8NfXl4.gif)

If this looks good, I can add tests. Also waiting on the final link to use for the bulk data.

Commit:
Adding cap to data export

Commit:
Adding tooltips

noahmanger commented:
Also need to add error handling for requests that turn out to be too large.

jmcarp commented on static/templates/tables/exportWidget.hbs:
```diff
@@ -1,7 +1,12 @@
+        Export this data
+    </button>
+    <div id="export-tooltip" role="tooltip" class="tooltip tooltip--under tooltip__content" aria-hidden="true">
+      <span>Exports are limited to 100,000 records— add filters to narrow results, or export bigger data sets with <a href="http://fec.gov/data/DataCatalog.do?cf=downloadable" target="_blank">FEC bulk data exporter</a>.</span>
```
Can we pass `DOWNLOAD_CAP` to the template so we don't have to hard-code the number here?

noahmanger replied:
:+1:

jmcarp commented on static/js/modules/tables.js:
```diff
@@ -485,12 +487,44 @@ DataTable.prototype.ensureWidgets = function() {
   this.hasWidgets = true;
 };
 
+DataTable.prototype.disableExport = function() {
```
These methods need tests. I'm fine doing a little manual QA and merging so we can test this with users if that would be useful, but we'll want tests before this gets into develop.

noahmanger replied:
Yep. I was going to add them if this looked good.

Commit:
Cancel downloads on 403 errors

Adds a simple method that to `close()` the download and issues a simple
browser alert message.

Wasn’t sure if textStatus was the right param to use here, but finding
documentation lacking online.

cc @jmcarp

Commit:
Removing hard-coded max count

cc @jmcarp

Commit:
Setting exportTooltip as a property of the table

Commit:
Semicolon party!

Commit:
Adding tests

Ping @jmcarp
